src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1245,12 +1245,30 @@ private bool IsAltSvcBlocked(HttpAuthority authority)
             {
                 lock (_altSvcBlocklist)
                 {
-                    return _altSvcBlocklist.Contains(authority);
+                    return _altSvcBlocklist.ContainsKey(authority);
                 }
             }
             return false;
         }
 
+
+        /// <summary>
+        /// Returns exception for badAuthority.
+        /// </summary>
+        /// <seealso cref="BlocklistAuthority" />
+        private Exception? ReturnBlocklistException(HttpAuthority badAuthority)
+        {
+            if (_altSvcBlocklist != null)
+            {
+                lock (_altSvcBlocklist)
+                {
+                    if (_altSvcBlocklist.TryGetValue(badAuthority, out Exception? ex))
+                        return ex;
+                }
+            }
+            return null;
+        }
```

Comment:
This logic should be made part of `IsAltSvcBlocked`. E.g.: `bool IsAltSvcBlocked(HttpAuthority authority, out Exception? reasonException)`